### PR TITLE
Random File Functions

### PR DESCRIPTION
**Pull request by schmamps**
- function to choose a random file (by extension) from a(n optionally specified) directory
- function to create a unique filename based on original

### CODE REVIEW

**Comment by coreyadkins** [comment]
Looks good to me. Ship it!

**Comment by Kaylotura** [comment]
Great. Ship it!

File: randomfile.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+def get_random_output_path(filename, dir = 'output'):
+    """ Appends a random string to a filename while preserving the extension
+
+    :return str: directory/basename.1234567890Ab.webm
+    """
+    basename, ext = splitext(filename)
+    rando = get_unique_str()
+
+    return join(dir, '{}/{}.{}{}'.format(dir, basename, rando, ext))
```
Comment (davidselassie) [inline randomfile.py]: These two functions are re-implementing Python's unique temporary file creation methods. Try using that!

``` py
from tempfile import mkstemp
mkstemp(dir='.', suffix='.png', prefix='dad')
```

Comment (schmamps) [inline randomfile.py]: This is a bit more than what I'd like, but it's implemented.

**Pull request merged**